app/models/king.rb

```diff
@@ -1,5 +1,48 @@
 class King < Piece
 
+  def valid_move?(new_x, new_y)
+    #checks if the move that the king is making is valid
+    if position_exist?(new_x, new_y) && pass_king_rules?(new_x, new_y)
+      return true
+    else
+      return false
+    end
+  end
+
+  def position_exist?(new_x, new_y)
+    #checks if position that king is moving to exists
+    if new_x < 0 || new_x > 7 || new_y < 0 || new_y > 7
+      return false
+    else
+      return true
+    end
+  end
+
+  def allowed_positions
```

Comment:
Ooh maybe we can do a check for something like if ((new_x == self.x_cord + 1) || (new_x == self.x_cord - 1)) && insert same pattern for y then the new x and y are good to go

There's no reason for us to list out all valid positions right? We just have to make sure the new x and y is within the square the king is allowed to travel? 

Just a midnight thought!

Sent from my iPhone

> On Jun 30, 2016, at 2:52 PM, Aleks Real <notifications@github.com> wrote:
> 
> In app/models/king.rb:
> 
> > +      return true
> > +    else
> > +      return false
> > +    end
> > +  end
> > +
> > +  def position_exist?(new_x, new_y)
> > +    #checks if position that king is moving to exists
> > +    if new_x < 0 || new_x > 7 || new_y < 0 || new_y > 7
> > +      return false
> > +    else
> > +      return true
> > +    end
> > +  end
> > +
> > +  def allowed_positions
> @ruishine that's a great idea but I have nooo idea how to do that lol I'm open to trying it out though, maybe we can pair program it?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>